test/integration/targets/ec2_instance/tasks/iam_instance_role.yml

```diff
@@ -10,7 +10,7 @@
 - block:
     - name: Create IAM role for test
       iam_role:
-        name: "{{ resource_prefix }}-test-policy"
+        name: "ansible-test-sts-{{ resource_prefix }}-test-policy"
```

Comment:
This probably was renamed so it would pass CI, but if I see a role with this name I'm going to assume it's being used for the sts_assume_role tests (since that's when I added the resource restriction). I wonder if the resource restriction should either be made more general, or if this should get its own resource restriction added.

Reply:
Up to you @s-hertel. The policies for automation only allow CreateRole on `ansible-test-sts-*` - I don't mind whether that's loosened so that we can tidy these tests.